test/tests/interfaces/pbs_queue_partition.py

```diff
@@ -0,0 +1,209 @@
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.interfaces import *
+
+
+@tags('multisched')
+class TestQueuePartition(TestInterfaces):
+
+    """
+    Test suite to test partition attr for queue
+    """
+
+    def create_queue_partition(
+            self,
+            q_type="e",
+            q_name="Q1",
+            partition="P1",
+            enable="True",
+            start="True"):
+        """
+        Common function to create queue with partition
+        :param q_type: "queue_type" attribute of queue object ,
+                    Defaults to "execution"
+        :type q_type:  str
+        :param q_name: name of the queue, Defaults to "Q1"
+        :type q_name: str
+        :param partition: "partition" attribute of queue object,
+                    Defaults to "P1"
+        :type partition: str
+        :param enable: "enabled" attribute of queue object, Defaults to "True"
+        :type enable: boolean
+        :param start: "started" attribute of queue object, Defaults to "True"
+        :type start: boolean
+        """
+        attr = {
+            'queue_type': q_type,
+            'enabled': enable,
+            'started': start,
+            'partition': partition}
+        self.server.manager(MGR_CMD_CREATE,
+                            QUEUE, attr,
+                            id=q_name, expect=True)
+
+    def test_set_unset_partition(self):
+        """
+        Test to set/unset the partition attribute of queue object
+        """
+        self.create_queue_partition()
+        self.server.manager(MGR_CMD_SET,
+                            QUEUE, {'partition': "P2"},
+                            id="Q1", expect=True)
+        # resetting the same partition value
+        self.server.manager(MGR_CMD_SET,
+                            QUEUE, {'partition': "P2"},
+                            id="Q1", expect=True)
+        self.server.manager(MGR_CMD_UNSET,
+                            QUEUE, 'partition',
+                            id="Q1", expect=True)
+
+    def test_set_partition_user_permissions(self):
+        """
```

Comment:
@hirenvadalia 
As mentioned in earlier comments, PTL returns proper error code in case of WINDOWS OS only since the shell on Windows does not manipulate the exit code returned by PBS commands like qmgr. But whereas in case of LINUX it returns (exit_code % 256) as return code.  Please look into the above suggestions made by us in earlier comments.

Also we checked the example you have given. Seems you executed that command on Windows only and hence you are getting the right error code by qmgr.